lib/translator/commands/array_stencil_command.rb

```diff
@@ -1,6 +1,85 @@
 module Ikra
     module Translator
         class CommandTranslator < Symbolic::Visitor
+
+            class NodeIndexModifier < AST::Visitor
+                # This visitor executes the check_parents_index function on every local variable
+                # If the local variable is the "stencil array" then its indices will get modified/corrected for the 1D array access
+
+                attr_reader :offsets
+                attr_reader :name
+
+                def initialize(name, offsets)
+                    @name = name
+                    @offsets = offsets
+                end
+
+                def visit_lvar_read_node(node)
+                    super(node)
+                    check_index(name, offsets, node)
+                end
+
+                def visit_lvar_write_node(node)
+                    super(node)
+                    check_index(name, offsets, node)
+                end
+
+
+                def check_index(name, offsets, node)
+                    if node.identifier == name
+                        # This is the array-variable used in the stencil
+
+                        send_node = node
+                        index_combination = []
+                        literal = true
+
+                        # Build the index off this access in index_combination
+                        for i in 0..offsets.first.first.size-1
```

Comment:
`offsets.first.first` is the number of dimension, right? Use `command.dimensions` instead (can pass an additional value to the visitor).